client/packages/invoices/src/Returns/SupplierListView/AppBarButtons.tsx

```diff
@@ -61,6 +64,14 @@ export const AppBarButtonsComponent: FC<{
               id: FnUtils.generateUUID(),
               supplierId: name?.id,
               supplierReturnLines: [],
+            }).then(invoiceNumber => {
+              navigate(
+                RouteBuilder.create(AppRoute.Replenishment)
+                  .addPart(AppRoute.SupplierReturn)
+                  .addPart(String(invoiceNumber))
+                  .build(),
+                { replace: false }
+              );
```

Comment:
Did you have a look into why this one didn't already have a navigate call here?

It's actually still using an old pattern, [calling the `navigate` from the API hook](https://github.com/msupply-foundation/open-msupply/blob/4097-browser-back-button-behaviour/client/packages/invoices/src/Returns/api/hooks/document/useInsertSupplierReturn.ts#L20)!

I think this is as good a time as any to update that! We can keep your navigate call here, and remove the one in the API hook. Currently the back function is getting a bit confused with both calls 😅

Reply:
Ah, yes this makes sense now that I know where to look!

Customer Return has the navigation through the API as well. I'd noticed that it didn't have navigation in AppBarButtons.tsx, however it did behave as expected. 
Would this be one to update now while I'm in it?

Reply:
Yes please! Great spotting 😁